Tests/MoSoAnalyticsTests/Mocks/MockBaseTracker.swift

```diff
@@ -0,0 +1,55 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import MoSoCore
+import MoSoAnalytics
+import XCTest
+
+class MockBaseTracker: BaseTracker {
+    var startExpectation: XCTestExpectation?
+    var stopExpectation: XCTestExpectation?
+    var impressionExpectation: XCTestExpectation?
+    var engagementExpectation: XCTestExpectation?
+
+    let mastodonAccountHandle: String
+    let mastodonAccountID: String
+
+    var engagementAction: EngagementAction = .none
+    var postID: String?
+    var recommendationID: String?
+    var associatedValue: String?
+    var additionalInfo: String?
+    var uiIdentifier: String?
+
+    init(session: MoSoSession) {
+        self.mastodonAccountID = session.user!.identifier
+        self.mastodonAccountHandle = session.user!.username
+    }
+
+    func start() {
+        startExpectation?.fulfill()
```

Comment:
curious: is this fulfill() being verified anywhere? if not, why?

Reply:
@cyndichin this is used in conjunction with the `waitForExpectations` call in some tests

Reply:
oh I thought we would need to specify, but I don't see it anywhere. how come we don't need to call it? 
```
baseTracker.startExpectation = expectation(description: "start call")
```

Reply:
The `start()` and `stop()` methods are not yet covered by these tests. This is because I encountered an unexpected crash while using the Glean testing api for these methods; I'd have to look a little deeper in these api and then add to these tests.